hackathon_site/event/views.py

```diff
@@ -1,18 +1,21 @@
 from rest_framework import generics, mixins
 
-from event.models import Profile
-from event.serializers import ProfileSerializer
+from event.models import User
+from event.serializers import UserSerializer
 
 
 class CurrentUserAPIView(generics.GenericAPIView, mixins.RetrieveModelMixin):
     """
     View to handle API interaction with the current user's Profile
     """
 
-    serializer_class = ProfileSerializer
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
 
     def get_object(self):
-        return generics.get_object_or_404(Profile.objects.all(), user=self.request.user)
+        queryset = self.get_queryset()
+
+        return generics.get_object_or_404(queryset, id=self.request.user.id)
```

Comment:
What's the point of doing 

    queryset = User.objects.all()
    queryset = self.get_queryset()
    return generics.get_object_or_404(queryset, id=self.request.user.id)

when you could just do 

    return generics.get_object_or_404(User.objects.all(), id=self.request.user.id)

Reply:
`queryset` is an attribute that serializers should have (or override `self.get_queryset()`), it was on omission on my part to hard code `User.objects.all()` into that method the first place.

Various generic reasons include:
- Some things, like the `RetreiveModelMixin`, use the default implementation of `.get_object` which filters the queryset on url kwargs: https://github.com/encode/django-rest-framework/blob/master/rest_framework/generics.py#L75
- It's helpful to have the queryset defined in one place in case you need to modify it. In this case, I should be doing a `.select_related("profile").prefetch_related("groups")`. If the view used it anywhere else, I wouldn't have to update it.
- Sometimes you might want to actually override `self.get_queryset()` instead, to eg return a different queryset depending on url arguments

Reply:
Actually in this case, prefetch_related doesn't help, but often it will save a lot of queries when dealing with many-to-many fields